test/testPartyAttachmentsAfterUpdate.js

```diff
@@ -0,0 +1,659 @@
+    if (file.attachment[prop] === undefined) {
+      throw new SriError({
+        status: 500,
+        errors: [
+          {
+            code: "file.attachment.is.missing.expected.property",
+            type: "ERROR",
+            message: `expected file.attachment object is missing property ${prop}`,
+          },
+        ],
+      });
+    }
+  })
+
+  if (file.resource === null || typeof file.resource !== 'object') {
+    throw new SriError({
+      status: 500,
+      errors: [
+        {
+          code: "file.resource.is.not.an.object",
+          type: "ERROR",
+          message: `expected resource.attachment to be an object`,
+        },
+      ],
+    });
+  }
+  const expectedResourceProperties = [
+    'href',
+    'key',
+  ]
+  expectedResourceProperties.forEach(prop => {
+    if (file.resource[prop] === undefined) {
+      throw new SriError({
+        status: 500,
+        errors: [
+          {
+            code: "file.resource.is.missing.expected.property",
+            type: "ERROR",
+            message: `expected file.attachment object is missing property ${prop}`,
+          },
+        ],
+      });
+    }
+  })
+}
+
+
+/**
+ * 
+ * @param {*} handleMultipleUploadsTogether 
+ * @param {*} uploadInSequence 
+ * @param {*} checkStoreAttachmentsReceivedList 
+ * @returns 
+ */
+function checkStoreAttachmentFactory(handleMultipleUploadsTogether, uploadInSequence, checkStoreAttachmentsReceivedList = undefined) {
+  if (handleMultipleUploadsTogether) {
+    return (files) => files.forEach(file => checkStoreAttachment(file))
+  } else {
+    if (uploadInSequence) {
+      return (file) => {
+        checkStoreAttachmentsReceivedList.push(file.fileObj.filename.substring(0,8));
+        checkStoreAttachment(file);
+      }
+    } else {
+      return checkStoreAttachment
+    }
+  }
+}
+
+function clearCheckStoreAttachmentsReceivedList(checkStoreAttachmentsReceivedList) {
+  if (checkStoreAttachmentsReceivedList) {
+    checkStoreAttachmentsReceivedList.length = 0; // clear the array
+  }
+}
+
+function verifyCheckStoreAttachmentsReceivedListOrder(checkStoreAttachmentsReceivedList) {
+  if (checkStoreAttachmentsReceivedList && checkStoreAttachmentsReceivedList.length > 0) {
+    const checkStoreAttachmentsReceivedListCopy = [...checkStoreAttachmentsReceivedList];
+    checkStoreAttachmentsReceivedListCopy.sort();
+    assert.deepStrictEqual(checkStoreAttachmentsReceivedList, checkStoreAttachmentsReceivedListCopy)
+  }
+}
+
+
+exports = module.exports = {
+  checkStoreAttachmentFactory,
+  factory: function (httpClient, type, checkStoreAttachmentsReceivedList) {
+    describe(type, function () {
+      describe("checkStoreAttachment", function () {
+        it("UPLOAD and COPY via upload", async () => {
+          try {
+            const body = {
+              type: "person",
+              name: "test user",
+              status: "active",
+            };
+            const resourceKey = uuid.v4();
```

Comment:
Creating the keys can be simplified to one statement with `Array.from()`, see eg https://bitbucket.org/vskovzw/content-api/src/main/test/testContentFilters.js#lines-165.